### ISSUE TITLE
ng-change gets triggered when a prohibited value is entered

### ISSUE THREAD

**Issue by min-che** [assisrafael/angular-input-masks#146]
I'm using ui-money-mask, and I have a function that gets called on ng-change.
When I try to enter a character, the value will (visually) not be changed, but ng-change gets triggered.

I'm using version 2.1.1

**Comment by assisrafael** [comment]
Can you write a test case that reproduce this behaviour?

**Comment by detinho** [comment]
I've made some tests and concluded that:

1. This behavior only happens if you clear the input and start typing invalid content, a letter for example. This way, ng-change will be fired as long as you type invalid content.
2. With the ui-number-mask the behavior is slight different: if you clear the input and type invalid content it fires ng-change, but already converts the model value to zero, so the ng-change will be fired only once.

I think the ui-number-mask behavior is correct, as the input value really changed from empty (undefined) to zero, so I've created a new pull request changing uiMoneyMask (https://github.com/assisrafael/angular-input-masks/pull/194).
